runtime/src/main/java/org/corfudb/runtime/GarbageInformer.java

```diff
@@ -156,10 +160,24 @@ public void submitGCTask() {
         }
     }
 
+    @VisibleForTesting
+    public void waitUntilAllTasksFinish() {
+        // waits until all task in the executor finish
+        drainExecutor.shutdown();
+        try {
+            if (!drainExecutor.awaitTermination(TERMINATION_WAIT_TIME.getSeconds(), TimeUnit.SECONDS)) {
+                drainExecutor.shutdownNow();
```

Comment:
You can put ShutdownNow() to the `finally` block.

Reply:
done.